hyperspy/utils/peakfinders2D.py

```diff
@@ -0,0 +1,572 @@
+# -*- coding: utf-8 -*-
+# Copyright 2007-2020 The HyperSpy developers
+#
+# This file is part of  HyperSpy.
+#
+#  HyperSpy is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+#  HyperSpy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with  HyperSpy.  If not, see <http://www.gnu.org/licenses/>.
+
+import numpy as np
+import scipy.ndimage as ndi
+from hyperspy.decorators import jit_ifnumba
+from skimage.feature import (
+        peak_local_max, blob_dog, blob_log, corner_peaks, match_template)
+import copy
+
+
+NO_PEAKS = np.array([[np.nan, np.nan]])
+
+
+@jit_ifnumba(nopython=True, cache=True)
+def _fast_mean(X):
+    """JIT-compiled mean of array.
+
+    Parameters
+    ----------
+    X : numpy.ndarray
+        Input array.
+
+    Returns
+    -------
+    mean : float
+        Mean of X.
+
+    Notes
+    -----
+    Used by scipy.ndimage.generic_filter in the find_peaks_stat
+    method to reduce overhead of repeated Python function calls.
+    See https://github.com/scipy/scipy/issues/8916 for more details.
+    """
+    return np.mean(X)
+
+
+@jit_ifnumba(nopython=True, cache=True)
+def _fast_std(X):
+    """JIT-compiled standard deviation of array.
+
+    Parameters
+    ----------
+    X : numpy.ndarray
+        Input array.
+
+    Returns
+    -------
+    std : float
+        Standard deviation of X.
+
+    Notes
+    -----
+    Used by scipy.ndimage.generic_filter in the find_peaks_stat
+    method to reduce overhead of repeated Python function calls.
+    See https://github.com/scipy/scipy/issues/8916 for more details.
+    """
+    return np.std(X)
+
+
+def clean_peaks(peaks):
+    """Sort array of peaks and deal with no peaks being found.
+
+    Parameters
+    ----------
+    peaks : numpy.ndarray
+        Array of found peaks.
+
+    Returns
+    -------
+    peaks : numpy.ndarray
+        Sorted array of found peaks.
+    NO_PEAKS : str
+        Flag indicating no peaks found.
+
+    """
+    if len(peaks) == 0:
+        return NO_PEAKS
+    else:
+        ind = np.lexsort((peaks[:,0], peaks[:,1]))
+        return peaks[ind]
+
+
+def find_local_max(z, min_distance=1, threshold_abs=None, threshold_rel=None,
+                   exclude_border=True, indices=True, num_peaks=np.inf,
+                   footprint=None, labels=None, num_peaks_per_label=np.inf):
```

Comment:
Can you clarify why you don't just pass `**kwargs` here?

Reply:
I did initially but then changed it to this for consistency with the log/dog methods already here. One advantage of explicitly stating these is that it's locally documented a bit better. I'm ambivalent about what is "right" though...

Reply:
OK. I'll defer to @francisco-dlp on the preferred approach here.

Reply:
@dnjohnstone's approach is certainly better in terms of usability, but it is brittle because any changes in the skimage function will break it or make it incomplete, increasing the maintenance burden. Hence, +1 for ``**kwargs``.

Reply:
@ericpre is there any reason I'm missing why you hadn't used **kwargs in other places already?